src/sidebar/components/moderation-banner.js

```diff
@@ -1,67 +1,108 @@
 'use strict';
 
-const annotationMetadata = require('../annotation-metadata');
-
-// @ngInject
-function ModerationBannerController(store, flash, api) {
-  const self = this;
+const { createElement } = require('preact');
+const classnames = require('classnames');
+const propTypes = require('prop-types');
 
-  this.flagCount = function() {
-    return annotationMetadata.flagCount(self.annotation);
-  };
+const annotationMetadata = require('../annotation-metadata');
+const useStore = require('../store/use-store');
+const { withServices } = require('../util/service-context');
 
-  this.isHidden = function() {
-    return self.annotation.hidden;
-  };
+/**
+ * Banner allows moderators to hide/unhide the flagged
+ * annotation from other users.
+ */
+function ModerationBanner({ annotation, api, flash }) {
+  // actions
+  const store = useStore(store => ({
+    hide: store.hideAnnotation,
+    unhide: store.unhideAnnotation,
+  }));
 
-  this.isHiddenOrFlagged = function() {
-    const flagCount = self.flagCount();
-    return flagCount !== null && (flagCount > 0 || self.isHidden());
-  };
+  const flagCount = annotationMetadata.flagCount(annotation);
 
-  this.isReply = function() {
-    return annotationMetadata.isReply(self.annotation);
+  const isHiddenOrFlagged = function() {
+    return flagCount !== null && (flagCount > 0 || annotation.hidden);
   };
 
   /**
    * Hide an annotation from non-moderator users.
    */
-  this.hideAnnotation = function() {
+  const hideAnnotation = () => {
     api.annotation
-      .hide({ id: self.annotation.id })
-      .then(function() {
-        store.hideAnnotation(self.annotation.id);
+      .hide({ id: annotation.id })
+      .then(() => {
+        store.hide(annotation.id);
       })
-      .catch(function() {
+      .catch(() => {
         flash.error('Failed to hide annotation');
       });
   };
 
   /**
    * Un-hide an annotation from non-moderator users.
    */
-  this.unhideAnnotation = function() {
+  const unhideAnnotation = () => {
     api.annotation
-      .unhide({ id: self.annotation.id })
-      .then(function() {
-        store.unhideAnnotation(self.annotation.id);
+      .unhide({ id: annotation.id })
+      .then(() => {
+        store.unhide(annotation.id);
       })
-      .catch(function() {
+      .catch(() => {
         flash.error('Failed to unhide annotation');
       });
   };
+
+  const toggleButtonProps = (() => {
+    const props = {};
+    if (annotation.hidden) {
+      props.onClick = unhideAnnotation;
+      props.title = 'Make this annotation visible to everyone';
+    } else {
+      props.onClick = hideAnnotation;
+      props.title = 'Hide this annotation from non-moderators';
+    }
+    return props;
+  })();
+
+  const bannerClasses = classnames('moderation-banner', {
+    'is-flagged': flagCount > 0,
+    'is-hidden': annotation.hidden,
+    'is-reply': annotationMetadata.isReply(annotation),
+  });
+
+  return (
```

Comment:
It may be a trifle easier to return `null` for the `!isHiddenOrFlagged` path here, e.g.

```
if (!isHiddenOrFlagged) {
  return null;
}

return (/* ...rendered stuff */);

```